internal/pkg/externalplugins/config.go

```diff
@@ -285,8 +293,25 @@ func (c *Configuration) TarsFor(org, repo string) *TiCommunityTars {
 	return &TiCommunityTars{}
 }
 
+// setDefaults will set the default value for the configuration of all plugins.
+func (c *Configuration) setDefaults() {
+	for i := range c.TiCommunityBlunderbuss {
+		c.TiCommunityBlunderbuss[i].setDefaults()
+	}
+}
+
+// setDefaults will set the default value for the config of blunderbuss plugin.
+func (c *TiCommunityBlunderbuss) setDefaults() {
+	if c.GracePeriodDuration == 0 {
+		c.GracePeriodDuration = defaultGracePeriodDuration
+	}
+}
+
 // Validate will return an error if there are any invalid external plugin config.
 func (c *Configuration) Validate() error {
+	// Defaulting should run before validation.
+	c.setDefaults()
```

Comment:
Maybe we shouldn't set defaults in this validate func, but it's OK for now. Could you please add a TODO to trace it?